### DISCUSSION TITLE
Integration of a Footsteps System in ALS-Refactored

### DISCUSSION THREAD

**Discussion by gustavorios2929** [Sixze/ALS-Refactored#433] [Ideas]
As a fan of ALS-Refactored, I'm curious if there are any plans to implement a footsteps sound system that would utilize the existing raycast mechanics for dynamic, surface-based audio, creating more immersive player feedback. It would be fantastic to have footstep sounds that match the various surfaces in-game.

Thank you!

**Comment by danieljcage** [comment]
> As a fan of ALS-Refactored, I'm curious if there are any plans to implement a footsteps sound system that would utilize the existing raycast mechanics for dynamic, surface-based audio, creating more immersive player feedback. It would be fantastic to have footstep sounds that match the various surfaces in-game.
> 
> Thank you!

It's already there. Add additional surface types to this data asset...
Plugins/ALS-Refactored/Content/ALS/Data/Footstep/FES_Als_Default.uasset

**Comment by gustavorios2929** [comment]
> > As a fan of ALS-Refactored, I'm curious if there are any plans to implement a footsteps sound system that would utilize the existing raycast mechanics for dynamic, surface-based audio, creating more immersive player feedback. It would be fantastic to have footstep sounds that match the various surfaces in-game.
> > Thank you!
> 
> It's already there. Add additional surface types to this data asset... Plugins/ALS-Refactored/Content/ALS/Data/Footstep/FES_Als_Default.uasset

Really?! That's amazing, thanks!
